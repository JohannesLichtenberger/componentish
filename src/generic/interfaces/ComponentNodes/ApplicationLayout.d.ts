import { ApplicationStyleProperty } from "./ApplicationPage";
import { GenericNode } from "./GenericNode";

export declare type LayoutType = "internal" | "external";
export interface ApplicationLayoutNode extends GenericNode {
    class: never;
    type: LayoutType;
    path?: string;
    styles?: ApplicationStyleProperty;
}
