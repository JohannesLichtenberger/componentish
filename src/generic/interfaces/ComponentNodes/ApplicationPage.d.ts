import { GenericNode } from "./GenericNode";
export interface ApplicationPageNode extends GenericNode {
    children: GenericNode[];
}
