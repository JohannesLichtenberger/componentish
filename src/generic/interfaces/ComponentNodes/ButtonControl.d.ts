import { RouteNode } from "../RouteNode";
import { GenericNode } from "./GenericNode";
export interface ButtonControlNode extends GenericNode {
    children: never;
    route: RouteNode;
}
