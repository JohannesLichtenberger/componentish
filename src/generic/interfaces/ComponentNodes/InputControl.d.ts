import { RouteNode } from "../RouteNode";
import { GenericNode } from "./GenericNode";
export interface InputControlNode extends GenericNode {
    children: never;
    value: string;
    placeHolder?: string;
    type?: inputTypes;
}

export declare type inputTypes = "text" | "mail" | "number" | "tel";
