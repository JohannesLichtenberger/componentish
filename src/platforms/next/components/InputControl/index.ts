import { indentation } from "../../../../generic/config/indentation";
import { staticImplements } from "../../../../generic/decorators/staticImplements";
import { InputControlNode } from "../../../../generic/interfaces/ComponentNodes/InputControl";
import { ComponentImport } from "../../../../generic/interfaces/transformer/ComponentImports";
import { GenericNodeTransformer } from "../../../../generic/interfaces/transformer/GenericNodeTransformer";

@staticImplements<GenericNodeTransformer>()
export class InputControl {
    public static generateMarkup(node: InputControlNode, level: number = 0) {
        const indent = indentation.repeat(level);
        const inputType = node.type || "text";
        const placeHolder = node.placeHolder ? `placeholder="${node.placeHolder}"` : "";
        return `${indent}<input type="${inputType}" ${placeHolder} id="${node.id}" />\n`;
    }
    public static getImports(_pageNode: InputControlNode): ComponentImport[] {
        return [
            {
                name: "InputControl",
                path: "@componnently/InputControl",
            },
        ];
    }
}

export default InputControl;
