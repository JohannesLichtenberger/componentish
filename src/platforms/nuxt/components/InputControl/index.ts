import { indentation } from "../../../../generic/config/indentation";
import { staticImplements } from "../../../../generic/decorators/staticImplements";
import { InputControlNode } from "../../../../generic/interfaces/ComponentNodes/InputControl";
import { ComponentImport } from "../../../../generic/interfaces/transformer/ComponentImports";
import { GenericNodeTransformer } from "../../../../generic/interfaces/transformer/GenericNodeTransformer";

@staticImplements<GenericNodeTransformer>()
export class InputControl {
    public static generateMarkup(node: InputControlNode, level: number = 0) {
        const indent = indentation.repeat(level);
        const inputType = node.type || "text";
        const placeHolder = node.placeHolder ? `placeholder="${node.placeHolder}"` : "";
        return `${indent}<input type="${inputType}" ${placeHolder} id="${node.id}" />\n`;
    }
    public static getImports(_pageNode: InputControlNode) {
        const imports: ComponentImport[] = [
            {
                name: "InputControl",
                path: "@componnently/InputControl",
            },
        ];
        return imports;
    }
}

export default InputControl;
