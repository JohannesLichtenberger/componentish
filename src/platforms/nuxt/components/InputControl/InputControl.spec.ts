import { InputControlNode } from "../../../../generic/interfaces/ComponentNodes/InputControl";
import { InputControl } from "../InputControl";

const mockNode = {
    controlType: "ActionControl",
    id: "ID_ActionControl_1_1",
    placeHolder: "Enter a text",
    value: "Test Value",
};
test("ButtonControl Snapshot", () => {
    const button = InputControl.generateMarkup(mockNode as InputControlNode);
    expect(button).toMatchSnapshot();
});
