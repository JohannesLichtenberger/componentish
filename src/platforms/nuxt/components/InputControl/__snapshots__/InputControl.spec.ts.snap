// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonControl Snapshot 1`] = `
"<input type=\\"text\\" placeholder=\\"Enter a text\\" id=\\"ID_ActionControl_1_1\\" />
"
`;
